certificate-rotation.md

```diff
@@ -0,0 +1,113 @@
+There are two parts to certificate rotation: generating the new certificates, and reloading them.
+
+Both have been improved, with the main difference being that we can now reload certificates without restarting a node.
+
+## Order of certificate regeneration
+
+Most of the complication in rotating certificates is proper handling of new CA certificates.
+It's important for all parts of the system (nodes and clients) to have the same CA certificates, or they may not be able to verify each other's client/node certificates.
+
+It is possible to have multiple CA certificates active at the same time, but not node or client certificates. Therefore, CA certificates should be generated and pushed first, followed by new node/client certificates using the new CA.
+
+Let's say we want to regenerate all certificates (CA, client, node), we should do the following:
+1. generate new CA certificate and combine it with the existing one
+1. push combined CA certificate to all nodes and reload certificates
+1. make combined CA available to all clients. We may need to wait some time (days/weeks) for all clients to make use of the new certificate
+1. generate and push new node/client certificates using the new CA certificate
+
+It is recommended to change CA certificates on a completely different schedule. eg: months before.
+**WARNING**: node certificates generated with the new CA will not be accepted by clients that do not yet have the new CA. Before all clients have the new CA cert, node certificates should be signed with the old CA (this is still a [TODO](https://github.com/cockroachdb/cockroach/blob/master/pkg/cli/cert.go#L91-L93))
```

Comment:
Are we going to handle that TODO before 1.0? If not, we might want to delay these docs.

Reply:
@mberhault Same question as before: If the combined CA has old CA as well, will this issue still occur?

Also, just to clarify, when we say "node certs should be signed with old CA", we mean use old CA key to generate new node cert and upload old CA cert to node, right?

Reply:
Similar answer as before: if the client only has the old CA, it will not be able to verify certificates signed with the new CA.
Think of the combined CA certificate as two completely separate certificate/key pairs that are joined together. If we supported it, we could use two completely different files instead of one file containing both CAs.